src/platform/test/CMakeLists.txt

```diff
@@ -46,3 +46,10 @@ if(BUILD_CINEMA)
 	add_test(cinema ${BINARY_NAME}-cinema -v)
 	install(TARGETS ${BINARY_NAME}-cinema DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT ${BINARY_NAME}-test)
 endif()
+
+if(BUILD_MTEST)
+	add_executable(${BINARY_NAME}-mtest ${CMAKE_CURRENT_SOURCE_DIR}/mtest-main.c)
```

Comment:
Given that this would wind up with a binary named mgba-mtest, I wonder if the second m is redundant. Maybe a name like mgba-test-harness, or mgba-rom-tester? It's kind of verbose, but is pretty explicit about what it is as a result.